WalletWasabi.Fluent/ViewModels/SearchBar/SearchBarViewModel.cs

```diff
@@ -29,21 +28,25 @@ public SearchBarViewModel(IObservable<IChangeSet<ISearchItem, ComposedKey>> item
 			.Filter(filterPredicate);
 
 		filteredItems
-			.Transform(item => item is ActionableItem i ? new AutocloseActionableItem(i, () => IsSearchListVisible = false) : item)
+			.Transform(
+				item => item is ActionableItem i
+					? new AutocloseActionableItem(i, () => IsSearchListVisible = false)
+					: item)
 			.Group(s => s.Category)
 			.Transform(group => new SearchItemGroup(group.Key, group.Cache))
 			.Bind(out _groups)
 			.DisposeMany()
 			.ObserveOn(RxApp.MainThreadScheduler)
 			.Subscribe();
 
-		_hasResults = filteredItems
+		HasResults = filteredItems
 			.Count()
 			.Select(i => i > 0)
-			.ToProperty(this, x => x.HasResults);
+			.Replay()
+			.RefCount();
```

Comment:
What are Replay and RefCount?

Reply:
**Replay** is an operator to avoid missing items from a subscriber that has subscribed after they have been emitted. 

Normally, we control when a subscriber subscribes to an observable, but in the case of Avalonia, it subscribes after the constructor of the ViewModel has been called. If any item is emitted before Avalonia has subscribed, the UI will not react, (nothing has happened in its point of view). To avoid this, we create an observable that broadcasts any item emitted from the beginning, not missing anything.

**RefCount** is an operator that auto-connects to a connectable observable (like those produced by Replay and Publish) and automatically disposes the underlying connection when the last subscribir unsubscribes. This is a "safety" mechanism to dispose the connection when it's no longer needed (nobody is listening to it).